capstone/src/test/java/com/google/step/servlets/LoginServletTest.java

```diff
@@ -0,0 +1,125 @@
+package com.google.step.servlets;
+
+import static org.junit.Assert.*;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.repackaged.com.google.gson.JsonObject;
+import com.google.appengine.repackaged.com.google.gson.JsonParser;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.meterware.httpunit.GetMethodWebRequest;
+import com.meterware.httpunit.WebRequest;
+import com.meterware.httpunit.WebResponse;
+import com.meterware.servletunit.ServletRunner;
+import com.meterware.servletunit.ServletUnitClient;
+import java.io.IOException;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.xml.sax.SAXException;
+
+public class LoginServletTest {
+    private final LocalServiceTestHelper helper =
+            new LocalServiceTestHelper();
+
+    @Before
+    public void setUp() {
+        helper.setUp();
+    }
+
+    @After
+    public void tearDown() {
+        helper.tearDown();
+    }
+
+    @Test
+    public void testIsLoggedIn() {
+        // Check if this works to set the user as logged in and not the admin
+        helper.setEnvIsAdmin(false).setEnvIsLoggedIn(true);
+        UserService userService = UserServiceFactory.getUserService();
+        assertTrue(userService.isUserLoggedIn());
+        assertFalse(userService.isUserAdmin());
+    }
+
+    @Test
+    public void testIsNotLoggedIn() {
+        // Check if it works to show the user is not logged in
+        helper.setEnvIsAdmin(false).setEnvIsLoggedIn(false);
+        UserService userService = UserServiceFactory.getUserService();
+        assertFalse(userService.isUserLoggedIn());
+        // Can't also check if the user is an admin because the method will throw an error since the
+        // user is not even logged in.
+    }
```

Comment:
Are these tests testing any code you wrote? If not, go ahead and remove them (the UserService should have its own tests that verify its behavior).

Reply:
Yeah I wrote them but you're right since it doesn't really test the functionality of my servlet.